File: src/lib/onboarding.ts
import { invoke } from "@tauri-apps/api/core";

export type OnboardingState = {
  completed: string[];
};

export function getOnboardingState(): Promise<OnboardingState> {
  return invoke<OnboardingState>("get_onboarding_state");
}

export function markOnboardingCompleted(id: string): Promise<OnboardingState> {
  return invoke<OnboardingState>("mark_onboarding_completed", { id });
}

File: src-tauri/src/commands/onboarding.rs
use crate::domain::onboarding::OnboardingState;
use crate::infra::onboarding_store;

#[tauri::command]
pub fn get_onboarding_state() -> Result<OnboardingState, String> {
    onboarding_store::load().map_err(|e| e.to_string())
}

#[tauri::command]
pub fn mark_onboarding_completed(id: String) -> Result<OnboardingState, String> {
    let mut state = onboarding_store::load().map_err(|e| e.to_string())?;
    if !state.completed.contains(&id) {
        state.completed.push(id);
        onboarding_store::save(&state).map_err(|e| e.to_string())?;
    }
    Ok(state)
}
